src/cloud.rs

```diff
@@ -222,11 +222,23 @@ impl<D: Device, P: Protocol, S: Socket, TS: TimeSource> GenericCloud<D, P, S, TS
 
     pub fn reset_own_addresses(&mut self) -> io::Result<()> {
         self.own_addresses.clear();
-        self.own_addresses.push(self.socket.address().map(mapped_addr)?);
-        if let Some(ref pfw) = self.port_forwarding {
-            self.own_addresses.push(pfw.get_internal_ip().into());
-            self.own_addresses.push(pfw.get_external_ip().into());
-        }
+        if self.config.advertise_addresses.len() > 0 &&
+            !self.config.listen.starts_with("ws://") {
+                // Force advertised addresses based on configuration istead
+                // of discovery. Note: Disables port forwarding
+                let port: u16 =
+                    try_fail!(self.config.listen.parse(), "Invalid port {}");
```

Comment:
Listen can be either `port` or `addr:port` (see `parse_listen` in net.rs)
I propose to change `parse_listen` to allow IP-only addresses and use the default port (`DEFAULT_PORT` in config.rs) whenever the address does not contain a colon and is not numeric.
This would add the flexibility to advertise a different port to what is specified in listen.

Reply:
refactored to use parse_listen